extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -236,24 +243,59 @@ public static Schema columnDefinitionsToAvroSchema(
         } else if (type == DateTime.class) {
             fass = base.longBuilder().prop(logicalTypeName, "timestamp-micros").endLong().noDefault();
         } else if (type == BigDecimal.class) {
-            final String precisionName = "precision";
-            final String scaleName = "scale";
-            String precision = null;
-            String scale = null;
-            if (colProps != null) {
-                precision = colProps.getProperty(colName + colNameToPropSeparator + precisionName);
-                scale = colProps.getProperty(colName + colNameToPropSeparator + scaleName);
-            }
-            if (precision == null || scale == null) {
-                throw new IllegalArgumentException(
-                        "When " + BigDecimal.class.getSimpleName() + " type columns are included, " +
-                                " column properties '" + precisionName + "' and '" + scaleName
-                                + "' for the given columns should be provided.");
+            final BigDecimalUtils.PrecisionAndScalePropertyNames propertyNames =
+                    new BigDecimalUtils.PrecisionAndScalePropertyNames(colName);
```

Comment:
This seems a little over teched rather than just telling the following function the name of the properties.  Or if you are going to have that; then the sets on 291 should also take the object instead of just doing it by hand.

Reply:
The issue is handling errors and giving the user meaningful exception text, while avoiding the repetition in the code for variable names and values of "precision" and "scale" in the two places this is used from.  I converged to this from having all this done manually so that I have a place to put the property names so that they can be used for the exception strings etc.  I would be happy to entertain a simpler way, I don't love it, but it is better than what I had before.

I don't understand you comment about line 291; we have two objects, one that contains the property names, and one that contains values.  You think we should have a function that takes those two and a properties object and set those?  The function would do exactly what is done in line 291, and it would be used from just one place (everything else in BigDecimalUtils is used from two places).

Reply:
What I am referring to is that on 291 you have all this manual code:
                    precision = precisionAndScale.precision;
                    scale = precisionAndScale.scale;
                    final Properties toSet;
                    if (colProps == null) {
                        toSet = new Properties();
                        colPropsMu.setValue(toSet);
                    } else {
                        toSet = colProps;
                    }
                    toSet.setProperty(propertyNames.precisionProperty, Integer.toString(precision));
                    toSet.setProperty(propertyNames.scaleProperty, Integer.toString(scale));

It seems like if you decided there should be a function to extract things from a property object, you should also have a function to set those from the property object.

It definitely feels like you have this division between your property name object, the holder object, and this code.  But then they are intertwined anyway.

Reply:
Yeah I see your point, plus the naming I don't think is the most happy.  Is hard for me to strike a balance for clarity without falling over to the "overly verbose (and lots of helper classes)" side.

I reworked the code, I think you will find it a better compromise.

Reply:
Charles and I discussed over slack, he considers this to have gone above the good enough threshold, so I am resolving this comment.